src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const intitialCopy = chatMessages.map((message) => {
+    return { ...message };
+  });
+  const [messagesList, setMessagesList] = useState(intitialCopy);
+
+  const updateLike = (entryId) => {
+    const newMessagesList = [];
+    for (const entry of messagesList) {
+      if (entry.id !== entryId) {
+        newMessagesList.push(entry);
+      } else {
+        const newEntry = {
+          ...entry,
+          liked: !entry.liked,
+        };
+        newMessagesList.push(newEntry);
+      }
+    }
+    setMessagesList(newMessagesList);
+  };
```

Comment:
This works well! As an alternative, can you consider what this would look like using `map`?